### PR TITLE
[receiver/elasticsearch] add jvm heap percentage usage metric

### PR DESCRIPTION
**Pull request by aboguszewski-sumo**
**Description:** 
A metric responsible for jvm heap usage in percents has been added.
Related comment: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/14635#issuecomment-1293521139

**Link to tracking Issue:** #14635

**Testing:** 
unit

**Documentation:** 
mdatagen

### CODE REVIEW

File: receiver/elasticsearchreceiver/metadata.yaml [hunk 1]
```diff
@@ -515,6 +515,14 @@ metrics:
       value_type: int
     attributes: []
     enabled: true
+  # other jvm metrics, disabled by default
+  jvm.memory.heap.percentage:
```
Comment (djaglowski) [inline receiver/elasticsearchreceiver/metadata.yaml:519]: I think this should probably be `jvm.memory.heap.utilization`, based on the [naming guidelines](https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter/exporterhelper#persistent-queue).

File: receiver/elasticsearchreceiver/scraper.go [hunk 2]
```diff
@@ -234,6 +234,8 @@ func (r *elasticsearchScraper) scrapeNodeMetrics(ctx context.Context, now pcommo
 
 		r.mb.RecordJvmThreadsCountDataPoint(now, info.JVMInfo.JVMThreadInfo.Count)
 
+		r.mb.RecordJvmMemoryHeapPercentageDataPoint(now, info.JVMInfo.JVMMemoryInfo.HeapUsedPercent)
```
Comment (djaglowski) [inline receiver/elasticsearchreceiver/scraper.go:237]: Can we move this up next to the other `JvmMemoryHeap` metrics?

File: receiver/elasticsearchreceiver/metadata.yaml (hunk 1 above)
Comment (aboguszewski-sumo) [reply receiver/elasticsearchreceiver/metadata.yaml]: The link is invalid (it leads to the persistent queue), did you mean [this](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/README.md#general-metric-semantic-conventions)?
I did change the name to `utilization`, but right in the guidelines stands `Utilization values are in the range [0, 1].`. Technically, this percentage is always in range `[0%,100%]` so it is the same as `[0, 1]`, but I'm not sure how strict that guideline is.

File: receiver/elasticsearchreceiver/scraper.go [hunk 3]
```diff
@@ -220,6 +220,8 @@ func (r *elasticsearchScraper) scrapeNodeMetrics(ctx context.Context, now pcommo
 		r.mb.RecordJvmMemoryHeapMaxDataPoint(now, info.JVMInfo.JVMMemoryInfo.MaxHeapInBy)
 		r.mb.RecordJvmMemoryHeapUsedDataPoint(now, info.JVMInfo.JVMMemoryInfo.HeapUsedInBy)
 		r.mb.RecordJvmMemoryHeapCommittedDataPoint(now, info.JVMInfo.JVMMemoryInfo.HeapCommittedInBy)
+		// Elasticsearch sends this data in percent, but we want to represent it as a number between 0 and 1, so we need to divide.
+		r.mb.RecordJvmMemoryHeapUtilizationDataPoint(now, float64(info.JVMInfo.JVMMemoryInfo.HeapUsedPercent)/100)
```
Comment (aboguszewski-sumo) [inline receiver/elasticsearchreceiver/scraper.go:224]: I think that we can trust that the division will be accurate (`info.JVMInfo.JVMMemoryInfo.HeapUsedPercent` is always an integer from range `[0;100]`), but correct me if I'm wrong.

File: receiver/elasticsearchreceiver/metadata.yaml (hunk 1 above)
Comment (aboguszewski-sumo) [reply receiver/elasticsearchreceiver/metadata.yaml]: Fixed.

File: receiver/elasticsearchreceiver/scraper.go (hunk 3 above)
Comment (djaglowski) [reply receiver/elasticsearchreceiver/scraper.go:224]: 👍 Looks correct to me.

**Review by djaglowski — APPROVED** [review]

**Pull request merged**